symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -386,4 +406,22 @@ private static Long getEpochMillis(Instant instant) {
   private <T> T executeAndRetry(String name, SupplierWithApiException<T> supplier) {
     return RetryWithRecovery.executeAndRetry(retryBuilder, name, supplier);
   }
+  
+  private File createTemporaryAttachmentFile(Attachment attachment) {
+    if (attachment == null) {
+      return null;
+    }
+    try {
+      String[] fileNameArrays = attachment.filename().split("\\.");
+      if (fileNameArrays.length < 2) {
+        throw new MessageCreationException("Invalid attachment's file name.");
+      }
+      File tempFile = File.createTempFile(fileNameArrays[0], "." + fileNameArrays[fileNameArrays.length - 1]);
+      tempFile.deleteOnExit();
```

Comment:
the file will be deleted only after the bot exits which could be never (if the bot is robust enough ::) 
If seems it could a risk in term of security, for instance if the bot is sending sensitive files, they would stay on disk for a long time in a temp folder that could be accessible to other users
I think we should find a way to not use a file at all (or maybe it is the only way to make it work for large files but in that case we should perhaps make it explicit that a folder is used for the files, making it appear in config for instance that the user is aware of the risk)
we could also try to delete the file after the call is made

Reply:
+1 for deleting the file after the call is made